src/app/people/people.component.ts

```diff
@@ -50,16 +51,24 @@ export class PeopleComponent implements OnInit {
   @Input() isEditingEnabled: boolean;
   @Output() onChanged = new EventEmitter<any>();
   @Output() onDelete = new EventEmitter<Person>();
-  displayedColumns: string[] = ["personDesc", "availability", "allocated", "unallocated", "assignmentCount", "commitFraction"];
   @ViewChild(MatSort) sort: MatSort;
-  
+
   constructor(public dialog: MatDialog) { }
 
   ngOnInit() {
   }
 
+  displayedColumns(): string[] {
+    let columnLabels = ['personDesc'];
+    if (this.people.find(p => p.location)) {
+      columnLabels.push('location');
+    }
+    columnLabels = columnLabels.concat(['availability', 'allocated', 'unallocated', 'assignmentCount', 'commitFraction']);
```

Comment:
Thinking about it further, I think you can just do this:

```suggestion
    columnLabels.push('availability', 'allocated', 'unallocated', 'assignmentCount', 'commitFraction');
```